src/_pytest/config/__init__.py

```diff
@@ -528,9 +528,15 @@ def _set_initial_conftests(
             if i != -1:
                 path = path[:i]
             anchor = absolutepath(current / path)
-            if anchor.exists():  # we found some file object
-                self._try_load_conftest(anchor, namespace.importmode, rootpath)
-                foundanchor = True
+            try:
+                if anchor.exists():  # we found some file object
+                    self._try_load_conftest(anchor, namespace.importmode, rootpath)
+                    foundanchor = True
+                print(str(anchor))
+            except OSError:
+                import pytest
+
+                raise pytest.UsageError(f"filename is too long: {anchor}")
```

Comment:
This does not seem correct to me: if the path given is raising `OSError` for some reason, we can assume in this case that it is not a valid file and follow the code path for the path not being found.

Something like:

```python
try:
    anchor_exists = anchor.exists()
except OSError:
    anchor_exists = False
if anchor_exists:
    self._try_load_conftest(anchor, namespace.importmode, rootpath)
    foundanchor = True
```

The simplest way to test this is I think using `monkeypatch` to mock `Path.exists` to raise an `OSError` if a specific file name is given (say `"longfile" * 200`), and pass that file during testing to ensure we are dealing with the `OSError` appropriately.

Reply:
But since it no longer raises the error, is there a need to test for it?

Reply:
> But since it no longer raises the error, is there a need to test for it?

Yes. The test should be there so we ensure the behavior works and keeps working in the future.

Without a test is possible that a future refactor removes the `OSError` handling (by mistake), and without a test it would go unnoticed, meaning that it would hit another user then. A test ensures this does not happen by mistake.